### PR TITLE
Proc creation susp curl fileupload modified

### PR DESCRIPTION
**Pull request by kidrek**
### Summary of the Pull Request

Modified rules to enable a case sensitive detection. Curl have two arguments "-f" and "-F" with different role.
I have implemented theses rules and had some false positives, 19k :( .

### CODE REVIEW

**Review by nasbench — APPROVED** [review]

**Comment by nasbench** [comment]
Thanks for the update @kidrek.

Next time please refrain from deleting anything in the template and only fills the sections if you have info for it :) (as its stated at the start).

Also when fixing FP please provide some examples to make the evaluation process easier.

**Comment by kidrek** [comment]
My bad @nasbench .
I'll do it next time.

File: rules/linux/process_creation/proc_creation_lnx_susp_curl_fileupload.yml [hunk 1]
```diff
@@ -21,17 +22,20 @@ logsource:
...
+    selection_cli:
+        - CommandLine|contains:
             - ' --form' # Also covers the "--form-string"
-            - ' -T '
             - ' --upload-file '
-            - ' -d '
             - ' --data '
             - ' --data-' # For flags like: "--data-ascii", "--data-binary", "--data-raw", "--data-urlencode"
-    condition: selection
+        - CommandLine|re: '\s-(F|T|d)\s' # We use regex to ensure a case sensitive argument detection
```
Comment (phantinuss) [inline rules/linux/process_creation/proc_creation_lnx_susp_curl_fileupload.yml:33]:
```suggestion
        - CommandLine|re: '\s-[FTd]\s' # We use regex to ensure a case sensitive argument detection
```
minor, but should be the "semantic" and "faster" regex for this job

File: rules/windows/process_creation/proc_creation_win_curl_fileupload.yml [hunk 2]
```diff
@@ -22,18 +22,17 @@ detection:
...
-            - ' -T '
             - ' --upload-file '
-            - ' -d '
             - ' --data '
             - ' --data-' # For flags like: "--data-ascii", "--data-binary", "--data-raw", "--data-urlencode"
-    condition: all of selection_*
-fields:
-    - CommandLine
-    - ParentCommandLine
+        - CommandLine|re: '\s-(F|T|d)\s' # We use regex to ensure a case sensitive argument detection
```
Comment (phantinuss) [inline rules/windows/process_creation/proc_creation_win_curl_fileupload.yml:30]:
```suggestion
        - CommandLine|re: '\s-[FTd]\s' # We use regex to ensure a case sensitive argument detection
```

**Review by phantinuss — APPROVED** [review]

**Pull request merged**